### PR TITLE
[WFLY-11328] EE Concurency Utilities hung-task-threshold / long-runni…

### PR DESCRIPTION
**Pull request by JiriOndrusek**
…ng-tasks do not work and are not implemented as explained

Issue: https://issues.jboss.org/browse/WFLY-11328

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:
- [ ] Pull Request title is properly formatted: `[WFLY-XYZ] Subject` or `WFLY-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue(s)
- [ ] Pull Request contains description of the issue(s)
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted

For bigger changes, major and minor component upgrades make sure your PR also meets following requirements:
- [ ] Pull Request requires a change to the documentation
- [ ] Documentation have been updated accordingly
- [ ] Tests were added to cover changes

For new features ensure as well:
- [ ] Analysis was done
- [ ] Test Plan has been done
- [ ] Tests were verified in advance

If you are not an active contributor of the WildFly project you can request sponsorship by one of the members to help guide you through the process.

### CODE REVIEW

**Comment by JiriOndrusek** [comment]
I'm aware that in PR's code there is a call of deprecated method (addDependency), but with it it is consistent with the rest of the code.

@emmartins May I ask for your opinion on this PR? It adds a small functionality into code mostly written by you...

**Review by kabir — CHANGES_REQUESTED** [review]
@JiriOndrusek one minor problem I can see.

@emmartins if you are still familiar can you please check it quickly?

**Comment by emmartins** [comment]
I can check it after lunch

File: ee/src/main/java/org/jboss/as/ee/concurrent/service/ManagedScheduledExecutorServiceService.java [hunk 1]
```diff
@@ -135,4 +164,34 @@ public ManagedScheduledExecutorServiceAdapter getValue() throws IllegalStateExce
...
+    private static class HungTaskAborter implements Runnable {
+
+        private final List<ManagedScheduledExecutorServiceImpl> managedScheduledExecutorServices = Collections.synchronizedList(new LinkedList<>());
+
+        public void addManagedScheduledExecutorService(ManagedScheduledExecutorServiceImpl managedScheduledExecutorService) {
+            this.managedScheduledExecutorServices.add(managedScheduledExecutorService);
+        }
+
+        public  void run() {
+            for(ManagedScheduledExecutorServiceImpl service : managedScheduledExecutorServices) {
```
[the remark this answers is not in the document]
Comment (kabir) [reply ee/src/main/java/org/jboss/as/ee/concurrent/service/ManagedScheduledExecutorServiceService.java]: @JiriOndrusek  It isn't. A synchronized list only guards the gets and adds, not the iteration. You should create a copy of the list in a synchronized block and then iterate the copy (which can happen outside the synchronized block).

[the remark this answers is not in the document]
Comment (JiriOndrusek) [reply ee/src/main/java/org/jboss/as/ee/concurrent/service/ManagedScheduledExecutorServiceService.java]: @kabir Yes, I've currently found it in docs, I see problem you are talking about. Thanks for pointing itout, I'll wait for @emmartins's response and then fix it.

**Comment by emmartins** [comment]
1. Any synchronisation here is not desirable, best effort checks would be fine
2. IMHO a timer interfering with all running threads every 10s is a very bad idea
That's regarding this specific PR, honestly I don't really like the idea of killing threads based on config no matter how implemented, I believe we should aim for a different non intrusive use of such functionality in RI, and fix docs.

**Comment by jamezp** [comment]
I'm closing this PR until it's decided what should be done here if anything.

**Pull request closed without merging**